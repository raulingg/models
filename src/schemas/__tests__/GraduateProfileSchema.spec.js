const mongoose = require('mongoose/browser');

const { Schema } = mongoose;

const { GraduateProfileSchema, UserSchema } = require('../')(mongoose);

describe('GraduateProfileSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, GraduateProfileSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values with HubspotReference', (done) => {
    const user = new mongoose.Document({}, UserSchema);
    const JobReferenceSchema = new Schema({
      createdBy: {
        type: Schema.Types.ObjectId,
        ref: 'User',
        required: true,
        index: true,
      },
      createdAt: {
        type: Date,
        default: Date.now,
      },
      url: {
        type: String,
        trim: true,
        required: true,
      },
    });
    const createdBy = new mongoose.Document({
      uid: 'xxxxxxxxxx',
      email: 'foo@bar.baz',
    }, UserSchema);

    const workReference = new mongoose.Document({
      createdBy: createdBy._id,
      url: 'https://www.hubspot.com/test',
    }, JobReferenceSchema);

    const graduateProfile = new mongoose.Document({
      user: user._id,
      jobReferences: [workReference],
    }, GraduateProfileSchema);

    graduateProfile.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
