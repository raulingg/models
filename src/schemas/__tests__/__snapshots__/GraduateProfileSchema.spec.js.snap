// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraduateProfileSchema should fail validation when fields missing 1`] = `
Object {
  "user": [ValidatorError: Path \`user\` is required.],
}
`;
